test/core/test/jest-mock.test.ts

```diff
@@ -363,6 +363,33 @@ describe('jest mock compat layer', () => {
     expect(obj.property).toBe(true)
   })
 
+  it('respyin on a spy resets the counter', () => {
```

Comment:
As long as this is working, it follows the expectation raised by https://github.com/vitest-dev/vitest/issues/7348#issuecomment-2613709450, so it looks safe to me. I cannot think of use cases where "spyOn multiple times" below causing issues.